project44/manifest-ui
# feat(components): adding focus states to inputs and buttons

# Description

DES-27 | Add focus states to focusable components.

We needed to add a more visual difference to focused elements. Adding a box shadow to all focusable components. We show the focus state on both mouse and keyboard focus events.

I tried to come up with a global way of applying these, however, each component has a subtly different focus state, background, border, or color change. This caused a global solution to be more cumbersome than helpful. We should definitely revisit this in the future to make focusable component composition easier. 

Before:
<img width="1108" alt="Screen Shot 2022-02-14 at 5 30 14 PM" src="https://user-images.githubusercontent.com/98107867/153964908-08387c0a-f711-46af-b98b-2a079c67b528.png">

After:
<img width="1034" alt="Screen Shot 2022-02-14 at 5 36 02 PM" src="https://user-images.githubusercontent.com/98107867/153964947-28ee78b7-fc23-4d6d-91c8-4b297933190d.png">

# 🔍  How to test this change locally

1. Run `yarn && yarn build && yarn dev`
1. Confirm changes by focusing on the button component.

# 🔮  Outstanding Questions

* Should we normalize to a single design for focus states? Does this help set user expectation for focus?
